swap/tests/alice_and_bob_refund_using_cancel_and_refund_command_timelock_not_expired_force.rs

```diff
@@ -0,0 +1,106 @@
+pub mod harness;
+
+use harness::alice_run_until::is_xmr_lock_transaction_sent;
+use harness::bob_run_until::is_btc_locked;
+use harness::SlowCancelConfig;
+use swap::protocol::alice::AliceState;
+use swap::protocol::bob::BobState;
+use swap::protocol::{alice, bob};
+
+#[tokio::test]
+async fn given_alice_and_bob_manually_force_cancel_when_timelock_not_expired_errors() {
+    harness::setup_test(SlowCancelConfig, |mut ctx| async move {
+        let (bob_swap, bob_join_handle) = ctx.bob_swap().await;
+        let swap_id = bob_swap.id;
+        let bob_swap = tokio::spawn(bob::run_until(bob_swap, is_btc_locked));
+
+        let alice_swap = ctx.alice_next_swap().await;
+        let alice_swap = tokio::spawn(alice::run_until(alice_swap, is_xmr_lock_transaction_sent));
+
+        let bob_state = bob_swap.await??;
+        assert!(matches!(bob_state, BobState::BtcLocked { .. }));
+
+        let (bob_swap, bob_join_handle) = ctx
+            .stop_and_resume_bob_from_db(bob_join_handle, swap_id)
+            .await;
+        assert!(matches!(bob_swap.state, BobState::BtcLocked { .. }));
+
+        let alice_state = alice_swap.await??;
+        assert!(matches!(
+            alice_state,
+            AliceState::XmrLockTransactionSent { .. }
+        ));
+
+        // Bob tries but fails to manually cancel
+        let is_outer_err = bob::cancel(bob_swap.id, bob_swap.bitcoin_wallet, bob_swap.db, true)
+            .await
+            .is_err();
+        assert!(is_outer_err);
```

Comment:
Normal `assert!`s produce pretty useless error messages.

Reply:
I changed it to match on an arbitrary outer error like this:

```
        // Bob tries but fails to manually cancel
        let result = bob::cancel(bob_swap.id, bob_swap.bitcoin_wallet, bob_swap.db, true)
            .await;
        assert!(matches!(result, Err(_)));
```

Hope that is more descriptive if it fails :)